Assignment Problems

NicholasMy opened the pull request:
## Proposed changes
--------------------
Add the assignment problems resource.

## Types of changes
--------------------
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist
--------------------
- [x] My changeset covers only what is described above (no extraneous changes)
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged downstream

Should resolve #25

MikeDrewitt commented:
Run `npm run format` please

MikeDrewitt commented:
I think the endpoints created here a bit wrong. The delete, update, and create are likely fine; but the list and detail endpoints may need to be either updated, or added to. 

Context for what this entity is going to be used for is important. Since all of these problems relate to a given assignment, we probably are never going to be wanting to request all assignment problems outside of the context of an assignment, right? So we may want to update that list endpoint to take an assignment id as a query parameter, or even in the route, or something so that where we actually use this data, it can retrieve a relevant list of stuff. (@jessehartloff @daviddob thoughts here?)

Similarly, depending on the context of where this is going to be used, we may not need a detail endpoint at all. Is there ever a context where we're going to want to request a single assignment problem? I can't think of any off the top of my head, but I might be missing something. So if we don't need to do that, would we want to just remove that endpoint in its entirety?

NicholasMy commented:
Thanks for the feedback, @MikeDrewitt, that makes sense. I've updated and tested the list endpoint to take an assignment id in the url (`/assignment-problems/{assignment-id}`) and only return assignment problems for that assignment id, and changed the detail endpoint to `/assignment-problems/detail/{id}`. I still think the detail endpoint may be useful, especially with the current way Autolab works. We don't necessarily need to clone its functionality, but it currently shows individual feedback per assignment problem, so I think the ability to retrieve details for an individual assignment problem by its id is valuable.
